client/index.js

```diff
@@ -50,6 +52,22 @@ const sox = (args = {}) => {
     })
   )
 
+  const sendWithOptions = curry((options, type, payload) =>
+    Async((reject, resolve) => {
+      const { timeout } = options
+      let sent = send(type, payload)
+
+      if (timeout != null) {
+        const err = error(type, { message: `Timeout after ${timeout}ms` })
+        const timer = setTimeout(() => reject(err), timeout)
+        const cleanup = tap(() => clearTimeout(timer))
+        sent = sent.bimap(cleanup, cleanup)
+      }
+
+      sent.fork(reject, resolve)
```

Comment:
@evilsoft This feels rather clunky to me, & i'm sure i missed something. I'm particularly worried that i'm not handling cancelled asyncs correctly.

Reply:
Relevant reading: https://mostly-adequate.gitbooks.io/mostly-adequate-guide/ch08.html#asynchronous-tasks

Reply:
In reading, I'm not sure I do either

Reply:
Head is exploding. Currently watching https://www.youtube.com/watch?v=O1Gu5b7rxbw

Reply:
Just watched the whole hour! Spoilers: no cancelling in it

Reply:
Related from Folktale:
https://folktale.origamitower.com/docs/v2.1.0/migrating/from-data.task/#cancelling-tasks

But I like `Async.timeout` a lot

Reply:
@evilsoft sorry for spam. Do you have a video / writeup on cancelling Asyncs manually? Not quite grokking it from the docs

Reply:
In crocks you cannot cancel an Async from the inside, breaks way too many laws.
BUT you can return a function that will be called when a fork is cancelled. When you call fork it will give you back a function. When that function is called it will cancel the forked flow. Will not resolve/reject and will not run an more Asyncs in a flow that are not already in flight or settled.

Reply:
Doing it the way we decided to in crocks keeps all side effects at the edge (as you call fork at the edge it makes sense for cancelling (like when some side effect pushes a cancel button, etc). So @mgreystone HAS to do it this way for his flow because he does not want to cancel but Reject after an amount of time.

Reply:
@evilsoft, should we be cancelling the inner `sent` Async, in addition to `reject`-ing the outer Async?